### PR TITLE
Document Git 27 changes

### LINKED ISSUE #4054: Git 27 changes
### What needs to be documented?

- https://docs.platform.sh/integrations/source/github.html#source-of-truth: It isn't possible to push to P.sh in this case.
- https://docs.platform.sh/development/variables/use-variables.html#use-provided-variables, adds the `PLATFORM_VENDOR` variable (`platformsh`/`upsun`; avail Build but _not_ Runtime) - to change the behavior of the build according to the vendor.
- "that made possible to restore a backup with more disk than the current disk of the environment needs  documentation, but I can't find the page" 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AnouckColson**
## Why

Closes #4054 

## What's changed

- Updated info on source integrations => now impossible to push directly to PSH/Upsun.
- Added `PLATFORM_VENDOR` variable to docs.
- Updated on how downsizing affects backup restoration.

### CODE REVIEW

File: sites/platform/src/development/variables/use-variables.md [hunk 1]
```diff
@@ -348,6 +348,7 @@ and at runtime.
 | `{{< vendor/prefix >}}_SOURCE_DIR`       | Yes   | No      | The path to the root directory of your code repository in the context of a running [source operation](../../create-apps/source-operations.md). The directory contains a writable copy of your repository that you can commit to during the operation. |
 | `{{< vendor/prefix >}}_TREE_ID`          | Yes   | Yes     | The ID of the tree the application was built from, essentially the SHA hash of the tree in Git. Use when you need a unique ID for each build. |
 | `{{< vendor/prefix >}}_VARIABLES`        | Some  | Some    | A base64-encoded JSON object with all user-defined project and environment variables that don't use a [prefix](./_index.md#variable-prefixes). The keys are the variable names and the values are the variable values. Availability during builds and at runtime depends on the settings for each variable. See how to [access individual variables](#access-variables-in-a-shell). |
+| `{{< vendor/prefix >}}_VENDOR`        | Yes  | No    |  | Allows you to change the behavior of the build according to the vendor. |
```
Comment (remilejeune2) [inline sites/platform/src/development/variables/use-variables.md:351]: We could specify what vendor means (upsun or platformsh). (Ignore that comment if it's a name we use in other places.)

**Review by remilejeune2 — APPROVED** [review]

File: sites/platform/src/development/variables/use-variables.md (hunk 1 above)
Comment (AnouckColson) [inline sites/platform/src/development/variables/use-variables.md:351]:
```suggestion
| `{{< vendor/prefix >}}_VENDOR`        | Yes  | No    | Allows you to change the behavior of the build according to the vendor ({{< vendor/name >}} or Upsun.) |
```

File: sites/upsun/src/development/variables/use-variables.md [hunk 2]
```diff
@@ -365,6 +365,7 @@ and at runtime.
 | `{{< vendor/prefix >}}_SOURCE_DIR`       | Yes   | No      | The path to the root directory of your code repository in the context of a running [source operation](../../create-apps/source-operations.md). The directory contains a writable copy of your repository that you can commit to during the operation. |
 | `{{< vendor/prefix >}}_TREE_ID`          | Yes   | Yes     | The ID of the tree the application was built from, essentially the SHA hash of the tree in Git. Use when you need a unique ID for each build. |
 | `{{< vendor/prefix >}}_VARIABLES`        | Some  | Some    | A base64-encoded JSON object with all user-defined project and environment variables that don't use a [prefix](./_index.md#variable-prefixes). The keys are the variable names and the values are the variable values. Availability during builds and at runtime depends on the settings for each variable. See how to [access individual variables](#access-variables-in-a-shell). |
+| `{{< vendor/prefix >}}_VENDOR`        | Yes  | No    |  | Allows you to change the behavior of the build according to the vendor. |
```
Comment (AnouckColson) [inline sites/upsun/src/development/variables/use-variables.md:368]:
```suggestion
| `{{< vendor/prefix >}}_VENDOR`        | Yes  | No    |  Allows you to change the behavior of the build according to the vendor ({{< vendor/prefix >}} or {{< vendor/psh_ref >}}. |
```

File: sites/platform/src/development/variables/use-variables.md [hunk 3]
```diff
@@ -348,6 +348,7 @@ and at runtime.
 | `{{< vendor/prefix >}}_SOURCE_DIR`       | Yes   | No      | The path to the root directory of your code repository in the context of a running [source operation](../../create-apps/source-operations.md). The directory contains a writable copy of your repository that you can commit to during the operation. |
 | `{{< vendor/prefix >}}_TREE_ID`          | Yes   | Yes     | The ID of the tree the application was built from, essentially the SHA hash of the tree in Git. Use when you need a unique ID for each build. |
 | `{{< vendor/prefix >}}_VARIABLES`        | Some  | Some    | A base64-encoded JSON object with all user-defined project and environment variables that don't use a [prefix](./_index.md#variable-prefixes). The keys are the variable names and the values are the variable values. Availability during builds and at runtime depends on the settings for each variable. See how to [access individual variables](#access-variables-in-a-shell). |
+| `{{< vendor/prefix >}}_VENDOR`        | Yes  | No    | Allows you to change the behavior of the build according to the vendor ({{< vendor/name >}} or Upsun.) |
```
Comment (AnouckColson) [inline sites/platform/src/development/variables/use-variables.md:351]:
```suggestion
| `{{< vendor/prefix >}}_VENDOR`        | Yes  | No    | Allows you to change the behavior of the build according to the vendor ({{< vendor/name >}} or Upsun). |
```

File: sites/upsun/src/development/variables/use-variables.md [hunk 4]
```diff
@@ -365,6 +365,7 @@ and at runtime.
 | `{{< vendor/prefix >}}_SOURCE_DIR`       | Yes   | No      | The path to the root directory of your code repository in the context of a running [source operation](../../create-apps/source-operations.md). The directory contains a writable copy of your repository that you can commit to during the operation. |
 | `{{< vendor/prefix >}}_TREE_ID`          | Yes   | Yes     | The ID of the tree the application was built from, essentially the SHA hash of the tree in Git. Use when you need a unique ID for each build. |
 | `{{< vendor/prefix >}}_VARIABLES`        | Some  | Some    | A base64-encoded JSON object with all user-defined project and environment variables that don't use a [prefix](./_index.md#variable-prefixes). The keys are the variable names and the values are the variable values. Availability during builds and at runtime depends on the settings for each variable. See how to [access individual variables](#access-variables-in-a-shell). |
+| `{{< vendor/prefix >}}_VENDOR`        | Yes  | No    |  Allows you to change the behavior of the build according to the vendor ({{< vendor/prefix >}} or {{< vendor/psh_ref >}}. |
```
Comment (AnouckColson) [inline sites/upsun/src/development/variables/use-variables.md:368]:
```suggestion
| `{{< vendor/prefix >}}_VENDOR`        | Yes  | No    |  Allows you to change the behavior of the build according to the vendor ({{< vendor/name >}} or {{< vendor/psh_ref >}}). |
```

**Review by chadwcarlson — APPROVED** [review]

**Pull request merged**